packages/dma-contracts/tasks/common/verification-utils.ts

```diff
@@ -100,3 +115,90 @@ export class ActionsDatabase {
     })
   }
 }
+
+export function validateActionHashes(
+  operationHashes: string[],
+  operationOptionals: boolean[],
+  actionDefinitions: ActionDefinition[],
+  actionsDatabase: ActionsDatabase,
+): ActionValidationResult {
+  let actionsValidated = true
+  let actionsErrorMessage: string | undefined = undefined
+
+  for (let actionIndex = 0; actionIndex < actionDefinitions.length; actionIndex++) {
+    if (
+      actionDefinitions[actionIndex].hash !== operationHashes[actionIndex] ||
+      actionDefinitions[actionIndex].optional !== operationOptionals[actionIndex]
+    ) {
+      actionsValidated = false
+
+      const actionDefinitionName = actionsDatabase.getActionName(
+        actionDefinitions[actionIndex].hash,
+      )
+      const operationHashName = actionsDatabase.getActionName(operationHashes[actionIndex])
+
+      actionsErrorMessage = `Action ${actionIndex} expected hash ${
```

Comment:
maybe aggregate the error message if more actions are invalid so you dont have to fix , run again, etc? Or thats unlikely?

Reply:
nice :)